### PR TITLE
mitmproxy 10.0.0

### PR DESCRIPTION
**Pull request by ZhongRuoyu**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

- [x] `resource` blocks have been checked for updates.
<details>
  <summary>release notes</summary>
  <pre>Changes: See [CHANGELOG.md](https://github.com/mitmproxy/mitmproxy/blob/main/CHANGELOG.md).

You can find the latest release packages at https://mitmproxy.org/downloads/.
</pre>
</details>

### CODE REVIEW

**Comment by chenrui333** [comment]
new build failure

```
error: failed to run custom build command for `mitmproxy v0.2.1 (/private/tmp/mitmproxy--mitmproxy-rs-20230805-9184-1upaypg/mitmproxy_rs-0.2.1/local_dependencies/mitmproxy)`
Caused by:
  process didn't exit successfully: `/private/tmp/mitmproxy--mitmproxy-rs-20230805-9184-1upaypg/mitmproxy_rs-0.2.1/target/release/build/mitmproxy-2164f44534e6d282/build-script-build` (exit status: 101)
  --- stderr
  thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Custom { kind: Uncategorized, error: "failed to invoke protoc (hint: https://docs.rs/prost-build/#sourcing-protoc): (path: \"/Users/brew/Library/Caches/Homebrew/cargo_cache/registry/src/index.crates.io-6f17d22bba15001f/protoc-bin-vendored-macos-x86_64-3.0.0/bin/protoc\"): Bad CPU type in executable (os error 86)" }', local_dependencies/mitmproxy/build.rs:12:6
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
💥 maturin failed
  Caused by: Failed to build a native library through cargo
  Caused by: Cargo build finished with "exit status: 101": `MACOSX_DEPLOYMENT_TARGET="11.0" PYTHON_SYS_EXECUTABLE="/opt/homebrew/Cellar/mitmproxy/10.0.0/libexec/bin/python" "cargo" "rustc" "--release" "--manifest-path" "/private/tmp/mitmproxy--mitmproxy-rs-20230805-9184-1upaypg/mitmproxy_rs-0.2.1/Cargo.toml" "--message-format" "json" "--lib" "--" "-C" "link-arg=-undefined" "-C" "link-arg=dynamic_lookup" "-C" "link-args=-Wl,-install_name,@rpath/mitmproxy_rs.abi3.so"`
Error: command ['maturin', 'pep517', 'build-wheel', '-i', '/opt/homebrew/Cellar/mitmproxy/10.0.0/libexec/bin/python', '--compatibility', 'off'] returned non-zero exit status 1
error: subprocess-exited-with-error

× Building wheel for mitmproxy_rs (pyproject.toml) did not run successfully.
│ exit code: 1
╰─> See above for output.
```

**Comment by chenrui333** [comment]
rebased after #138904

**Comment by chenrui333** [comment]
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Custom { kind: Uncategorized, error: "failed to invoke protoc (hint: https://docs.rs/prost-build/#sourcing-protoc): (path: \"/Users/brew/Library/Caches/Homebrew/cargo_cache/registry/src/index.crates.io-6f17d22bba15001f/protoc-bin-vendored-macos-x86_64-3.0.0/bin/protoc\"): Bad CPU type in executable (os error 86)" }', local_dependencies/mitmproxy/build.rs:12:6
```

File: Formula/mitmproxy.rb [hunk 1]
```diff
@@ -94,9 +104,9 @@ class Mitmproxy < Formula
     sha256 "af598ed32d6ae86f1b747b82783958b1a4ab8f617b06fe68795c7f026abbdcad"
   end
 
-  resource "mitmproxy-wireguard" do
-    url "https://files.pythonhosted.org/packages/46/71/5529c5c0eefdee81dedce09deb76578dd416deec38e10b70a111d57322fd/mitmproxy_wireguard-0.1.23.tar.gz"
-    sha256 "b0f7b44ef9b0601307c122c5fe1ce57368c2fc9330097ec576984a0d640b4727"
+  resource "mitmproxy-rs" do
+    url "https://files.pythonhosted.org/packages/db/1a/8c1678c625fd4e4c000b4c2e696d0121b66e1348bfa7b03755bb21441758/mitmproxy_rs-0.2.1.tar.gz"
```
Comment (mhils) [inline Formula/mitmproxy.rb:108]: Bumping this should fix the CI issues if protoc is installed.on the builder. :)

**Comment by mhils** [comment]
Thanks, @ZhongRuoyu! I've filed https://github.com/stepancheg/rust-protoc-bin-vendored/issues/1 to get macOS ARM support for protoc-bin-vendored. Independently, we'll also upstream your patch so that homebrew does not need to do extra work to build. 👍

**Comment by ZhongRuoyu** [comment]
Thank you @mhils! Now our CI is good. 🎉 I also opened mitmproxy/mitmproxy_rs#85 and added an explanation in the PR description.

**Review by branchv — APPROVED** [review]

**Pull request merged**